R/expect-request-header.R

```diff
@@ -82,5 +82,13 @@ expect_request_header <- function(expr,
     }
   }
 
-  httr2::with_mock(header_mocker, expr)
+  with_mock(header_mocker, expr)
+}
+
+with_mock <- function(mock, code) {
+  if (httr2_1.0.0) {
+    httr2::with_mocked_responses(mock, code)
```

Comment:
I think this needs to be `getFromNamespace` if I am to submit an update to CRAN before httr2 lands; this currently raises a check warning.